### ISSUE TITLE
Message size limit enforced incorrectly

### ISSUE THREAD

**Issue by cgillum** [Azure/azure-functions-durable-extension#189]
Currently we use the following code to check for maximum message sizes for function inputs and outputs:

```csharp
// String payloads in Azure Storage are encoded in UTF-32.
int payloadSizeInKB = (int)(Encoding.UTF32.GetByteCount(serializedJson) / 1024.0);
if (payloadSizeInKB > MaxPayloadSizeInKB)
{
    throw new ArgumentException(
        string.Format(
            "The UTF-32 size of the JSON-serialized payload must not exceed 60 KB. The current payload size is {0:N0} KB.",
            payloadSizeInKB));
}
```

This is actually wrong. Azure Storage string are encoded in **UTF-16**, so our current size check is too strict. Instead, it should be something like this:

```csharp
// String payloads in Azure Storage are encoded in UTF-16.
int payloadSizeInKB = (int)(Encoding.Unicode.GetByteCount(serializedJson) / 1024.0);
if (payloadSizeInKB > MaxPayloadSizeInKB)
{
    throw new ArgumentException(
        string.Format(
            "The UTF-16 size of the JSON-serialized payload must not exceed 60 KB. The current payload size is {0:N0} KB.",
            payloadSizeInKB));
}
```
While we won't enforce message sizes when we enable [large messages support](https://github.com/Azure/azure-functions-durable-extension/issues/26), we'll still want to make sure we're calculating message sizes correctly since we don't want to upload to blob (or compress) unless we actually need to.

**Comment by tohling** [comment]
Fixed in commit https://github.com/Azure/azure-functions-durable-extension/pull/190/commits/87a55582a1a946a3c904f46a324a1a7a2a2893b1

**Comment by cgillum** [comment]
This fix has been included in the [Beta 3.1 release (v1.2.1-beta3)](https://github.com/Azure/azure-functions-durable-extension/releases/tag/v1.2.1-beta3).

**Comment by known-as-bmf** [comment]
There are still UTF32 references in the code

https://github.com/Azure/azure-functions-durable-extension/blob/dev/src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs#L1074
https://github.com/Azure/azure-functions-durable-extension/blob/dev/src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs#L1104

Is this desired since encoding was changed to Unicode in some other places ?
